From 673d5036ac9130304e5ce5460f6188573f698383 Mon Sep 17 00:00:00 2001
From: Simon Dinkin <simonx.dinkin@intel.com>
Date: Mon, 16 Jan 2017 15:59:43 +0200
Subject: [PATCH] send eapols from wpa_supplicant to the driver as vendor event

use the existing LTQ_NL80211_VENDOR_SUBCMD_TX_EAPOL vendor command
to send the eapols to the driver.

Signed-off-by: Simon Dinkin <simonx.dinkin@intel.com>
---
 wpa_supplicant/wpas_glue.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/wpa_supplicant/wpas_glue.c b/wpa_supplicant/wpas_glue.c
index f84c8b9..fbfd5e8 100644
--- a/wpa_supplicant/wpas_glue.c
+++ b/wpa_supplicant/wpas_glue.c
@@ -111,11 +111,7 @@ static int wpa_ether_send(struct wpa_supplicant *wpa_s, const u8 *dest,
 	}
 #endif /* CONFIG_TESTING_OPTIONS */
 
-	if (wpa_s->l2) {
-		return l2_packet_send(wpa_s->l2, dest, proto, buf, len);
-	}
-
-	return -1;
+	return wpa_drv_hapd_send_eapol(wpa_s, dest, buf, len, 0, wpa_s->own_addr, 0);
 }
 #endif /* IEEE8021X_EAPOL || !CONFIG_NO_WPA */
 
-- 
2.10.1

